test/e2e/manifest_test.go

```diff
@@ -377,12 +377,16 @@ add_compression = ["zstd"]`), 0o644)
 		session = podmanTest.Podman([]string{"manifest", "annotate", "--annotation", "hello=world,withcomma", "--arch", "bar", "foo", imageListARM64InstanceDigest})
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(ExitCleanly())
+		session = podmanTest.Podman([]string{"manifest", "annotate", "--index", "--annotation", "left=right", "foo"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
 		session = podmanTest.Podman([]string{"manifest", "inspect", "foo"})
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(ExitCleanly())
 		Expect(session.OutputToString()).To(ContainSubstring(`"architecture": "bar"`))
-		// Check added annotation
+		// Check added annotations
 		Expect(session.OutputToString()).To(ContainSubstring(`"hello": "world,withcomma"`))
+		Expect(session.OutputToString()).To(ContainSubstring(`"left": "right"`))
```

Comment:
Doesn't both annotation are set on different levels? I feel like such a test must make sure they are set in the correct places not just that they exists anywhere.

Reply:
Yes, they are.  Reworked the comparison logic.